### ISSUE TITLE
How to check and log errors during payments

### ISSUE THREAD

**Issue by IlPizza79** [VivaPayments/API#1553]
Hi, I'm developing a custom Prestashop module for simple checkout integration and I need to check errors during payments.

Does Viva Wallet provide a built-in error log for payment transaction? Or do I need to create my own error log manually?

Thanks in advance.

**Comment by matthewrgourd** [comment]
Hi @IlPizza79 while making a test transaction in your browser you should be able to see any [rejection reason](https://developer.vivawallet.com/api-reference-guide/error-codes/#rejection-reasons) in the Network tab when you inspect the page (Google Chrome). Is that relevant to what you meant?

**Comment by IlPizza79** [comment]
Hi, what I meant is that I need a way to track errors on REAL transaction, not test.
Some of my customers said that they where unable to pay with Viva Wallet simple chechout, and I would like to track the error code and know the reason of the failed transaction.

So I simply would like to know if Viva Wallet provides a built-in error log or if I need to log errors by myself in my custom module.

Thanks again.

**Comment by matthewrgourd** [comment]
The answer is yes, what I mentioned above applies to the live environment. I was talking about performing a test in production.

**Comment by IlPizza79** [comment]
Sorry, I don't understand... the answer is "yes" to which question between the two I made?
Unfortunately my english is not that great and maybe my question was a bit unclear...
let me try again :-)

- Does Viva Wallet provides a built-in error log where I can see errors for REAL transaction made by my customers? Y/N
- Do I need to log errors by myself in my custom module? Y/N

Thanks again :-)

**Comment by matthewrgourd** [comment]
Hi @IlPizza79 the answer is 'yes' to the first question and therefore 'no' to the second question. Thanks!

**Comment by IlPizza79** [comment]
Hi, so the asnswer to "Does Viva Wallet provides a built-in error log where I can see errors for REAL transaction made by my customers?" is "yes", and that's great :-)

Then, can you please tell me how and where can I see those errors?
Thanks again!
